tools/runner/reporter.go

```diff
@@ -217,6 +217,18 @@ func (tcr *TestCaseReporter) Duration() time.Duration {
 	return tcr.endTime.Sub(tcr.startTime)
 }
 
+// AddProperty adds a key-value property to the test case.
+func (tcr *TestCaseReporter) AddProperty(key, value string) {
+	property := &xunit.Property{
```

Comment:
You can make this shorter by testing for nil in the beginning and then appending the property in one line instead of creating a separate "property" variable.